src/components/CaretakerFilters.tsx

```diff
@@ -90,27 +91,28 @@ const CaretakerFilters: React.FC<CaretakerFiltersProps> = ({
           showSearch={false}
           placeholder={t("size")}
           onChange={(value) =>
-            onAnimalFiltersChange(animalType, { attributes: { SIZE: value as AnimalSize[] } })
+            onAnimalFiltersChange(animalType, { attributes: { SIZE: value } })
           }
           value={animalFilters[animalType]?.attributes?.SIZE || []}
-          options={[
-            { value: "SMALL", label: t("small") },
-            { value: "MEDIUM", label: t("medium") },
-            { value: "BIG", label: t("big") }
-          ]}
+          notFoundContent={t("noData")}
+          options={store.animal.getAttributeValues(animalType, "SIZE").map((size) => ({
+            value: size,
+            label: t(size.toLowerCase())
+          }))}
         />
         <Select
           mode="multiple"
           showSearch={false}
           placeholder={t("sex")}
           onChange={(value) =>
-            onAnimalFiltersChange(animalType, { attributes: { SEX: value as AnimalSex[] } })
+            onAnimalFiltersChange(animalType, { attributes: { SEX: value } })
           }
           value={animalFilters[animalType]?.attributes?.SEX || []}
-          options={[
-            { value: "MALE", label: t("male") },
-            { value: "SHE", label: t("she") }
-          ]}
+          notFoundContent={t("noData")}
+          options={store.animal.getAttributeValues(animalType, "SEX").map((sex) => ({
```

Comment:
tutaj jak ustawiasz to SEX to ustawiasz pole do wyboru w filtrach nie? Bo jak cos to na backendzie specjalnie robilem strukture pod to zeby latwo mozna bylo rozszerzyc na wiecej rozdzajow cech, wiec jakby dalo sie jakos generowac formularz filtowania na podstawie posiadanych cech w jakiejs tablicy zamiast na sztywno to pewnie lepiej. Ale moge tez nie rozumiec kodu

Reply:
Moge tak zrobić zeby filtry były dynamiczne w zależności od liczby cech które zwrócisz.